test/Rings-conformance-tests.jl

```diff
@@ -21,6 +21,113 @@ function equality(a::T, b::T) where T <: AbstractAlgebra.NCRingElement
    end
 end
 
+function test_mutating_op_like_zero(f::Function, f!::Function, a)
+  A = deepcopy(a)
+  a = f!(a)
+  @test equality(a, f(A))
+  a = deepcopy(A)
+end
+
+function test_mutating_op_like_neg(f::Function, f!::Function, a, b)
+  A = deepcopy(a)
+  B = deepcopy(b)
+  a = f!(a, b)
+  @test equality(a, f(B))
+  @test b == B
+  a = deepcopy(A)
+  b = deepcopy(B)
+
+  a = f!(a)
+  @test equality(a, f(A))
+  a = deepcopy(A)
+end
+
+function test_mutating_op_like_add(f::Function, f!::Function, a, b, c)
+  A = deepcopy(a)
+  B = deepcopy(b)
+  C = deepcopy(c)
+
+  a = f!(a, b, c)
+  @test equality(a, f(B, C))
+  @test b == B
+  @test c == C
+  a = deepcopy(A)
+  b = deepcopy(B)
+  c = deepcopy(C)
+
+  a = f!(a, a, b)
+  @test equality(a, f(A, B))
+  @test b == B
+  a = deepcopy(A)
+  b = deepcopy(B)
+
+  a = f!(a, b, a)
```

Comment:
In practice this won't always work; think of `divexact!(::QQMatrix, ::QQMatrix, ::ZZRingElem)`

I am also not sure how useful it is in practice.

Here is my impression (I might be missing something, though): if `b` and `a` have the same type, this swapped test is redundant; while if tf they have different types, they might be harmful; but if the caller truly wants to test both orders (e.g. both `mul!(::ZZRingElem, ::ZZRingElem, ::Integer)` and `mul!(::ZZRingElem, ::Integer, ::ZZRingElem)` they can just call this function twice?

Reply:
it is not redundant, as it tests a different combination of aliasing between the arguments of `f!`

I'd rather add another function for `*_like_divexact`